python-lib/timeseries_preparation/preparation.py

```diff
@@ -67,9 +82,65 @@ def prepare_timeseries_dataframe(self, dataframe):
             log_message_prefix = f"Sampling {self.max_timeseries_length} last records, obtained"
             self._log_timeseries_lengths(dataframe_prepared, log_message_prefix=log_message_prefix)
 
+        if self.timeseries_identifiers_names:
+            if self.timeseries_identifiers_values:
+                self._check_identifiers_values(dataframe_prepared)
+            else:
+                self.timeseries_identifiers_values = (
+                    dataframe_prepared[self.timeseries_identifiers_names].drop_duplicates().to_dict("records")
+                )
+
         return dataframe_prepared
 
+    def check_schema_from_dataset(self, dataset_schema):
+        dataset_columns = [column["name"] for column in dataset_schema]
+        expected_columns = (
+            [self.time_column_name]
+            + (self.target_columns_names or [])
+            + (self.timeseries_identifiers_names or [])
+            + (self.external_features_columns_names or [])
+        )
+        if not set(expected_columns).issubset(set(dataset_columns)):
+            raise ValueError(f"Dataset of historical data must contain the following columns: {expected_columns}")
```

Comment:
I can move the check done in the TrainingSession class (it checks that the target and external features columns are numerical) to the TimeseriesPreparator class. This way we make sure it's numerical without ever enforcing the dss type (this plugin never looks at the dss types because it uses infer_with_pandas=True). 
The date column is already checked and for the identifiers column, it's even better than checking the types because we also check the values.

Reply:
Done that in 3b2fee3

Reply:
alright!